packages/sign-client/src/controllers/engine.ts

```diff
@@ -286,6 +280,20 @@ export class Engine extends IEngine {
       controller: selfPublicKey,
     };
     await this.client.session.set(sessionTopic, session);
+    try {
+      await this.sendRequest({
+        topic: sessionTopic,
+        method: "wc_sessionSettle",
+        params: sessionSettle,
+        throwOnFailedPublish: true,
+      });
+    } catch (error) {
+      this.client.logger.error(error);
+      // if the publish fails, delete the session and throw an error
+      this.client.session.delete(sessionTopic, getSdkError("USER_DISCONNECTED"));
```

Comment:
Do we also need to unsubscribe from `sessionTopic` in this case to undo https://github.com/WalletConnect/walletconnect-monorepo/pull/4105/files#diff-04e05e19b9270d1267b68d324e16997e4465d6879324fe4427a6194c00410eabR269 or is that undesirable?

Reply:
good catch! Adding..